regression/test.pl

```diff
@@ -64,9 +64,21 @@ ($)
   return @data;
 }
 
-sub test($$$$$$) {
-  my ($name, $test, $t_level, $cmd, $ign, $dry_run) = @_;
+sub test($$$$$$$) {
+  my ($name, $test, $t_level, $cmd, $ign, $dry_run, $defines) = @_;
   my ($level, $input, $options, $grep_options, @results) = load("$test");
+  my @keys = keys %{$defines};
+  foreach my $key (@keys) {
+    my $value = $defines->{$key};
+    $options =~ s/(\$$key$|\$$key )/$value /g;
+  }
+  if (scalar @keys) {
+    foreach my $word (split(/\s/, $options)) {
+      if ((substr($word, 0, 1) cmp '$') == 0) {
+        print "\n  Warning: Key \"$word\" not replaced\n";
```

Comment:
I still think the message should mention the `-D` option, otherwise a user not aware of that option will wonder why the script is complaining on $. But that's a non-blocking comment (and the only one).